extensions/amp-analytics/0.1/amp-analytics.js

```diff
@@ -413,13 +413,40 @@ export class AmpAnalytics extends AMP.BaseElement {
       return false;
     }
 
-    const props = this.config_['optout'].split('.');
+    // TODO(leeandrew1693): This block is to support backwards compatibility
+    // for GTM. Remove this block once GTM updates.
+    if (typeof this.config_['optout'] === 'string') {
+      const props = this.config_['optout'].split('.');
+      let k = this.win;
+      for (let i = 0; i < props.length; i++) {
+        if (!k) {
+          return false;
+        }
+        k = k[props[i]];
+      }
+      // The actual property being called is controlled by vendor configs only
+      // that are approved in code reviews. User customization of the `optout`
+      // property is not allowed.
+      return k();
+    }
+
+    const elementId = this.config_['optout']['id'];
+    if (elementId && this.win.document.getElementById(elementId)) {
+      return true;
+    }
+
+    const functionName = this.config_['optout']['function'];
```

Comment:
i thought we agreed to not support the old method? @zikas to keep me honest.

Reply:
I believe that we've agreed on having deprecation strategy. 
https://github.com/ampproject/amphtml/issues/21163

Once this change goes live, we'll update GTM code and fallback could be safely removed.

Reply:
yep, that's what I meant. So we don't need the code here, since we already have code addressing the backward compatibility issue.

Reply:
We still need both opt out implementations since we are supporting both the id optout and the function optout moving forward. My understanding was that we would only remove the fallback of 
```
'optout': 'function.name'
```
but still keep
```
'optout': {
  'function': 'function.name',
}
```

I would remove this backwards compatible code once GTM updates its code:
```
'optout': 'function.name'
```

Reply:
any reason we need to keep 

```
'optout': {
  'function': 'function.name',
}
```

even after GTM is fully migrated?

Reply:
For background, we provide extensions for 5 browsers -- Chrome, Opera, IE, Safari, and Firefox. CSP affects optout on 2 of them (Safari and Firefox). Since the extensions are installed on the user's browser, we do not have full control over whether or not an update is accepted. As such, there is a non-zero chance that after an update is sent, some users won't have the latest version of the extension. This is a problem that client teams (apps, android/ios OS, etc) all run into.

In the scenario that we do  migrate off the 'function.name' way of optout, we cannot guarantee that all customers have the latest update since there will always the long tail customers who don't update the extension. This means that we would break their optout functionality.

Reply:
I see. In that case, can I suggest we have a new field for the new feature and keep the old one?

```
optout: "_gaUserPrefs.ioo"
optoutElementId: "__gaOptOutExtension"
``` 

This will make the backward compatibility code simpler and future deprecation easier.

Reply:
FYI I'll be out for vacation. Handing it over to @zhouyx for review.

Reply:
That solution sounds good to me. I've updated the pull request to do as you suggested

Reply:
@zhouyx could you take a look?